tests/pipelines/test_pipelines.py

```diff
@@ -1802,6 +1804,42 @@ def test_pipe_same_device_id_offload(self):
         sd.maybe_free_model_hooks()
         assert sd._offload_gpu_id == 5
 
+    def test_wrong_model(self):
+        tokenizer = CLIPTokenizer.from_pretrained("hf-internal-testing/tiny-random-clip")
+        with self.assertRaises(ValueError) as error_context:
+            _ = StableDiffusionPipeline.from_pretrained(
+                "hf-internal-testing/diffusers-stable-diffusion-tiny-all", text_encoder=tokenizer
+            )
+
+        assert "is of type" in str(error_context.exception)
+        assert "but should be" in str(error_context.exception)
```

Comment:
We're now using warning, but for this case, `CLIPTokenizer` in `text_encoder` we still get a `ValueError` later on from here
https://github.com/huggingface/diffusers/blob/63243406ba5510c10d5cac931882918ceba926f9/src/diffusers/pipelines/pipeline_utils.py#L893-L897

So it's a little inconsistent and needs further testing to determine which other cases this already applies to.